cmake/protobuf.cmake

```diff
@@ -0,0 +1,51 @@
+include(GNUInstallDirs)
+
+# Set protobuf options
+set(Protobuf_USE_STATIC_LIBS ON)
+set(Protobuf_MSVC_STATIC_RUNTIME OFF)
+set(protobuf_BUILD_TESTS OFF)
+set(protobuf_BUILD_PROTOC_BINARIES ON)
+
+# Fetch protocolbuffers_protobuf
+message(STATUS "")
+message(STATUS "Fetching protocolbuffers/protobuf")
+FetchContent_Declare(protobuf
+  DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+  URL "https://github.com/protocolbuffers/protobuf/archive/refs/tags/v3.16.0.tar.gz"
```

Comment:
Just a curiosity here. Do you think we can set a "library version variable" or something like that that syncs this version of protobuf to the python package requirement in "python/requirements.txt"? Maybe this is sciencefinction but maybe you know some tricks.

Reply:
Absolutely no clues on that, @mberaha, so sorry!

Reply:
According to ChatGPT, it is possible to parse a file to get a version variable. I just don't understand what we actually need from the requirements.txt: there's no protobuf in it